src/main/java/org/elasticsearch/index/query/RangeQueryBuilder.java

```diff
@@ -434,4 +276,123 @@ protected String parserName() {
         return RangeQueryParser.NAME;
     }
 
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query query = null;
+        MapperService.SmartNameFieldMappers smartNameFieldMappers = parseContext.smartFieldMappers(this.fieldname);
+        if (smartNameFieldMappers != null) {
+            if (smartNameFieldMappers.hasMapper()) {
+                FieldMapper mapper = smartNameFieldMappers.mapper();
+                if (mapper instanceof DateFieldMapper) {
+                    if ((from instanceof Number || to instanceof Number) && timeZone != null) {
+                        throw new QueryParsingException(parseContext,
+                                "[range] time_zone when using ms since epoch format as it's UTC based can not be applied to [" + this.fieldname
+                                        + "]");
+                    }
+                    DateMathParser forcedDateParser = null;
+                    if (this.format  != null) {
+                        forcedDateParser = new DateMathParser(Joda.forPattern(this.format), DateFieldMapper.Defaults.TIME_UNIT);
+                    }
+                    DateTimeZone dateTimeZone = null;
+                    if (this.timeZone != null) {
+                        dateTimeZone = DateTimeZone.forID(this.timeZone);
+                    }
+                    query = ((DateFieldMapper) mapper).rangeQuery(from, to, includeLower, includeUpper, dateTimeZone, forcedDateParser, parseContext);
+                } else  {
+                    if (timeZone != null) {
+                        throw new QueryParsingException(parseContext, "[range] time_zone can not be applied to non date field ["
+                                + this.fieldname + "]");
+                    }
+                    //LUCENE 4 UPGRADE Mapper#rangeQuery should use bytesref as well?
+                    query = mapper.rangeQuery(from, to, includeLower, includeUpper, parseContext);
+                }
+
+            }
+        }
+        if (query == null) {
+            query = new TermRangeQuery(this.fieldname, BytesRefs.toBytesRef(from), BytesRefs.toBytesRef(to), includeLower, includeUpper);
+        }
+        query.setBoost(boost);
+        if (queryName != null) {
+            parseContext.addNamedQuery(queryName, query);
+        }
+        return query;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (this.fieldname == null || this.fieldname.isEmpty()) {
+            validationException = QueryValidationException.addValidationError("field name cannot be null or empty.", validationException);
```

Comment:
Yes it sure does make sense. Just wondering if we are changing the behaviour of the query while refactoring. Can you test what the query currently does in master? Maybe it simply falls apart with no field name?